### PR TITLE
fix editing, adding, duplicating node frontend

### LINKED ISSUE #104: frontend:editing a node property is broken

### PR DESCRIPTION
**Pull request by Emmanuel-Develops**
closes #104

### CODE REVIEW

File: frontend/src/contexts/network-context.tsx [hunk 1]
```diff
@@ -25,10 +25,11 @@ export const networkContext = React.createContext<NetworkContext>(null!);
 const newNode = {
   id: "0",
   data: {
-  name: "new node",
-  label:"new node",
-  size: 10,
+    id: "0",
```
Comment (0tuedon) [inline frontend/src/contexts/network-context.tsx:28]: i see duplicates keys (id).

**Review by 0tuedon — COMMENTED** [review]
Nice work, left some comments.

File: frontend/src/contexts/network-context.tsx (hunk 1 above)
Comment (Emmanuel-Develops) [reply frontend/src/contexts/network-context.tsx:28]: The `data` property, has a field `id` which is the `id` here. I made the `id` field mandatory (it was an optional type previously). The formatting makes it look like a duplicate, its a field under data.

Discussions on wether the field is redundant is a different matter. (I'm indifferent)
I think if we want separation of concern between node properties for the canvas (id, type, position) and node data itself, we can leave the second `id` field to reference the node.

There's a specific instance where separation of concern can simplify the editing fnc but its not a priority at the moment so we can leave the `id` as is for now.

File: frontend/src/contexts/node-flow-context.tsx [hunk 2]
```diff
@@ -68,82 +63,81 @@ export const NodeGraphFlowProvider = ({
...
   const updateNodeInfo = (nodeProperty: any, value: any) => {
-    if (!nodeInfo) return
-    const duplNode = {...nodeInfo}
+    if (!nodeInfo) return;
+    const duplNode = JSON.parse(JSON.stringify(nodeInfo));
     //@ts-ignore partia will come back to it
-    duplNode.data[nodeProperty]  = value
-    setNodeInfo(duplNode)
-  }
+    duplNode.data[nodeProperty] = value;
```
Comment (Extheoisah) [inline frontend/src/contexts/node-flow-context.tsx:117]: I guess we no longer need the ts ignore since node data is no longer a partial

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**